cmd/hccore/hccore.go

```diff
@@ -0,0 +1,136 @@
+// command line interface to developing and testing holochain applications
+
+// Copyright (C) 2013-2017, The MetaCurrency Project (Eric Harris-Braun, Arthur Brock, et. al.)
+// Use of this source code is governed by GPLv3 found in the LICENSE file
+//---------------------------------------------------------------------------------------
+// command line interface to developing and testing holochain applications
+
+package main
+
+import (
+  "fmt"
+  "github.com/metacurrency/holochain/cmd"
+  "github.com/urfave/cli"
+  "os"
+
+  "bytes"
+)
+
+const (
+  defaultPort = "4141"
+)
+
+func getCurrentDirectoryOrExit() (string) {
+  dir, err := cmd.GetCurrentDirectory()
+  if err != nil {
+    cmd.Die("HC: hccore.go: getCurrentDirectory: could not find current directory. Weird. Exitting")
+  }
+
+  return dir
+}
+
+var debug bool
+var rootPath, devPath, name string
+var fromWhichSourceDirectory string
+  var noQuestions, compile bool
+
+func setupApp() (app *cli.App) {
+  app           = cli.NewApp()
+  app.Name      = "hccore"
+  app.Usage     = "tools for Holochain Core developers"
+  // app.Version = fmt.Sprintf("0.0.0 (holochain %s)", holo.VersionStr)
+  app.Flags     = []cli.Flag{
+
+  }
+
+  app.Commands  = []cli.Command{
+    {
+      Name:               "fromLocalFilesystem",
+      // Aliases:    []string{"branch"},
+      Usage:              "install a Holochain Core from a local directory (defaults to .)",
+      Flags:              []cli.Flag{
+          cli.StringFlag{
+            Name:          "sourceDirectory",
+            Usage:         "path to source files containing checked out git branch, defaults to current directory",
+            Value:         getCurrentDirectoryOrExit(),
+            Destination:   &fromWhichSourceDirectory,
+          },
+      },
+      Subcommands:    []cli.Command{
+        {
+          Name:           "install",
+          // Aliases:    []string{"branch"},
+          Usage:          "install the version of Holochain Core in '.' onto the host system",
+          Flags:          []cli.Flag{
+              cli.BoolFlag{
+                Name:          "noQuestions",
+                Usage:         "once the files are made available to Golang, should we compile them?",
+                Destination:   &noQuestions,
+              },
+              cli.BoolFlag{
+                Name:          "compile",
+                Usage:         "once the files are made available to Golang, should we compile them?",
+                Destination:   &compile,
+              },
+          },
+          Action:         func(c *cli.Context) error {
+              fmt.Printf  ("HC: core.fromLocalFilesystem.install: installing from        : %v\n",     fromWhichSourceDirectory)
+
+              err := os.Chdir(fromWhichSourceDirectory)
+              if err != nil {
+                fmt.Printf("HC: core.fromLocalFilesystem.install: could not change dir to: %v\n",     fromWhichSourceDirectory)
+                os.Exit(1)
```

Comment:
I think if we use: return fmt.Errorf(...) here urfave/cli doesthe os.Exit(1) for us.  At least that's how I did things in hc.